script.js

```diff
@@ -1,92 +1,138 @@
-    }
-    var allGrades = $('.grade');
-    for (var i = 0; i < allGrades.length; i++) {
-        var creditUnits = document.getElementsByClassName('credit-unit');
-        var grades = document.getElementsByClassName('grade');
-
-        if (grades[i].value == 'a' || grades[i].value == 'A') {
-            var gradeValue = 5;
-            creditPoints.push(Number(creditUnits[i].value) * gradeValue);
-        }
-        else if (grades[i].value == 'b' || grades[i].value == 'B') {
-            var gradeValue = 4;
-            creditPoints.push(Number(creditUnits[i].value) * gradeValue);
-        }
-        else if (grades[i].value == 'c' || grades[i].value == 'C') {
-            var gradeValue = 3;
-            creditPoints.push(Number(creditUnits[i].value) * gradeValue);
-        }
-        else if (grades[i].value == 'd' || grades[i].value == 'D') {
-            var gradeValue = 2;
-            creditPoints.push(Number(creditUnits[i].value) * gradeValue);
-        }
-        else if (grades[i].value == 'e' || grades[i].value == 'E') {
-            var gradeValue = 1;
-            creditPoints.push(Number(creditUnits[i].value) * gradeValue);
-        }
-        else if (grades[i].value == 'f' || grades[i].value == 'F') {
-            var gradeValue = 0;
-            creditPoints.push(Number(creditUnits[i].value) * gradeValue);
-        }
-        else {
-            swal('Invalid Entry');
 
             // reset all variables 
             creditPoints = [];
-            totalCreditUnit = 0;
+            allCreditUnitInputs = [];
             cgpa = 0;
 
-            return;
-        }
-    }
+        });
+
+        $('#downloadbtn').click(() => {
+            for (var i = 0; i < document.getElementsByTagName('input').length; i++) {
+                if (document.getElementsByTagName('input')[i].value == '' ) {
+                    swal('Fill in inputs and calculate your cgpa first');
+                    return;
+                }
+            }
+
+            for (var i = 0; i <= numberOfCourses; i++) {
 
-    var total = 0;
-    for (var i = 0; i < creditPoints.length; i++) {
-        total += creditPoints[i];
-    }
+                if (i === numberOfCourses) {
+                    document.querySelector('tbody').insertAdjacentHTML('beforeend', `<tr><td>Your CGPA is ${result.toFixed(2)}</td></tr>`)
+                    document.querySelector('tr').style.fontWeight = 'bold'
+                } else {
+                    document.querySelector('tbody').insertAdjacentHTML('beforeend', '<tr></tr>')
+                    var courses = document.getElementsByClassName('course');
+                    var creditUnits = document.getElementsByClassName('credit-unit');
+                    var grades = document.getElementsByClassName('grade');
+                    document.querySelector('tbody').lastElementChild.insertAdjacentHTML('beforeend', `<td>${courses[i].value}</td><td>${creditUnits[i].value}</td><td>${grades[i].value.toUpperCase()}</td>`)
```

Comment:
Could we add more columns
Like GP(Grade Point of the Alphabet grade), QP = (GP * CU)